### PR TITLE
Implement panning support

### LINKED ISSUE #595: Touchpad: Allow two-finger panning
The graph can now be zoomed via touchpad gestures, but panning isn't possible yet, which can be a bit confusing since that's the behavior other apps/platforms use.

### PR DESCRIPTION
**Pull request by Sjoerd1993**
Implements drag to pan with two-fingers on the touchpad, or using horizontal/vertical scrolls. 
To zoom in with the scroll wheel a ctrl modifier now has to be used, resulting in the same behaviour as applications like Inkscape and LibreOffice.

Closes #595 

Still WIP, as this needs to be cleaned up and will probably not pass linting, haven't even checked the linter locally but thought I'd put this up before I call it a day. Also there's some slight bugs in the inverse scale at least, and need to tweak the values a bit.

### CODE REVIEW

**Comment by Sjoerd1993** [comment]
This now completely works.

This now also rewrites the zoom calculation logic, using the `utilities.get_value_at_fraction` method instead so that everything just gets treated with a linear logic. This greatly simplifies that calculation and reduces a lot of replication as it now all uses this single calculation points. Deviating axes can all just use that logic, making this more expandable.

File: src/canvas.py [hunk 1]
```diff
@@ -142,14 +144,25 @@ def _set_mouse_fraction(self, event):
...
         self.zoom(scale)
 
+    def _on_pan_gesture(self, _event_controller, x, y):
+        if self.get_application().get_ctrl() is False:
+            for ax in self.axes:
+                xmin, xmax, ymin, ymax = \
+                    self._calculate_pan_values(ax, x, y)
+                ax.set_xlim(xmin, xmax)
+                ax.set_ylim(ymin, ymax)
+            self.queue_draw()
```
Comment (cmkohnen) [inline src/canvas.py:160]: A problem with both panning and zooming now is, that it doesn't get added to the view history. A proper implementation for that can be part of another pr however.

**Review by cmkohnen — CHANGES_REQUESTED** [review]

File: src/canvas.py [hunk 2]
```diff
@@ -111,7 +108,7 @@ def __init__(self, application, style_params):
         color_rgba.alpha = 0.3
         self.rubberband_fill_color = utilities.rgba_to_tuple(color_rgba, True)
         # Reference is created by the toolbar itself
-        _DummyToolbar(self)
+        self.toolbar = _DummyToolbar(self)
```
[the remark this answers is not in the document]
Comment (Sjoerd1993) [reply src/canvas.py:111]: This is there from when I experimented a bit with this toolbar, it shouldn't have been here as the final code hence it not being used either. It's also redundant anyway as it's saved in figure.canvas.toolbar.

File: src/application.py [hunk 3]
```diff
@@ -160,6 +169,10 @@ def do_activate(self):
             window.get_stack_switcher_box().prepend(stack_switcher)
             window.set_title(self.props.name)
             self.set_window(window)
+            controller = Gtk.EventControllerKey.new()
```
[the remark this answers is not in the document]
Comment (Sjoerd1993) [reply src/application.py:172]: If I recall correctly there was a reason for this but will take a look when I've got that time.

File: src/canvas.py (hunk 1 above)
Comment (Sjoerd1993) [reply src/canvas.py:160]: Both gestures seem to emit end signals, so this should be easy to implement. Will take a look.

File: src/canvas.py [hunk 4]
```diff
@@ -172,31 +185,42 @@ def zoom(self, scaling=1.15, respect_mouse=True):
             ))
         self.queue_draw()
 
+    @staticmethod
+    def _calculate_pan_values(ax, x_panspeed, y_panspeed):
+        xmin, xmax = min(ax.get_xlim()), max(ax.get_xlim())
+        ymin, ymax = min(ax.get_ylim()), max(ax.get_ylim())
+        x_scale = scales.to_int(ax.get_xscale())
+        y_scale = scales.to_int(ax.get_yscale())
```
[the remark this answers is not in the document]
Comment (Sjoerd1993) [reply src/canvas.py:193]: Will take a look, I think it's possible for the limits at least to utilize the internals

**Review by cmkohnen — APPROVED** [review]
